indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -148,7 +161,12 @@ public static Configuration injectSystemProperties(Configuration conf)
   {
     for (String propName : System.getProperties().stringPropertyNames()) {
       if (propName.startsWith("hadoop.")) {
-        conf.set(propName.substring("hadoop.".length()), System.getProperty(propName));
+        String prop = propName.substring("hadoop.".length());
```

Comment:
Can this logic be combined with the other method doing a similar thing, maybe like a "maybeAddJobProperty" method that both things can call?